oldsite/static/js/toggle.js

```diff
@@ -0,0 +1,12 @@
+const currentPagePath = window.location.pathname;
+const currentPageName = currentPagePath.substring(currentPagePath.lastIndexOf('/') + 1);
+console.log(currentPageName);
+
+const pageToggle = document.getElementById('page-toggle');
+let currentPage = currentPageName;
+
+pageToggle.addEventListener('click', function() {
```

Comment:
I would put this in an `if (pageToggle) { ... }` so that there will be no errors if such an element does not exist.

Reply:
Thanks @felixfontein I was thinking about how to do that because toggling between the old and new docsites was "jumpy" because pages aren't 1:1 between the two. My js-fu isn't so strong (as you can tell from that dirty code I wrote) and I wasn't getting the if clause to work. TBH it's still not working but this has helped me think of a way to solve the jumpy-ness and avoid stub pages.

For pages on the new site that don't exist on the old, we just won't add a toggle. We aren't going to add the user journeys to the oldsite and keep those cards so a toggle doesn't really make sense.

For pages on the old site that we'll either drop or rename we can do a copy as part of the build process, although maybe that is a little dangerous if someone commits and the next build overwrites. I don't think we really need to worry about that too much though.

Reply:
For renamed pages, you could also add a data field to the toggle HTML element with the name of the correct destination. Then the JS could use that value to send the user to the right place.

Regarding the toggle: you should probably add a listener to the checkbox `<input>` and not the `<label>`, so that if a user clicks on the checkbox (or uses some other mechanism to toggle it, say in a text-based browser that runs JS) it will also toggle, and not just if the user clicks on the label.

Reply:
snap. that's a great point about the listener. I changed it from a button to a slider and garbled that.